test/extended/testdata/gssapi/scripts/gssapi-tests.sh

```diff
@@ -0,0 +1,258 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo "Running ${TEST_NAME}"
+
+cd "${OS_ROOT}"
+source hack/lib/init.sh
+
+function cleanup() {
+    out=$?
+    # get the jUnit output file into a workable state in case we crashed in the middle of testing something
+    os::test::junit::reconcile_output
+    # check that we didn't mangle jUnit output
+    os::test::junit::check_test_counters
+    exit $out
+}
+
+trap "cleanup" EXIT
+os::test::junit::declare_suite_start "${TEST_NAME}"
+
+# Client has no GSSAPI libs and server is GSSAPI only
+# Everything fails
+# Errors do NOT mention Kerberos
+
+if [[ "${CLIENT}" = 'CLIENT_MISSING_LIBS' && "${SERVER}" = 'SERVER_GSSAPI_ONLY' ]]; then
+
+    os::cmd::expect_failure_and_text 'oc login' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_text 'oc whoami' 'system:anonymous'
+
+    os::cmd::expect_failure_and_text 'oc login -u user1' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user1'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p wrongpassword' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_failure_and_text 'oc login -u user2 -p password' 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user2'
+
+    os::cmd::expect_failure_and_text "oc login -u 'user3@${REALM}'" 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user3'
+
+    os::cmd::expect_failure_and_text "oc login -u 'user4@${REALM}' -p wrongpassword" 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user4'
+
+    os::cmd::expect_failure_and_text "oc login -u 'user5@${REALM}' -p password" 'Login failed \(401 Unauthorized\)'
+    os::cmd::expect_failure_and_not_text 'oc whoami' 'user5'
+
+fi
+
+# Client has uncofigured GSSAPI libs and server is GSSAPI only
+# Everything fails
+# Errors mention Kerberos
+
+if [[ "${CLIENT}" = 'CLIENT_HAS_LIBS' && "${SERVER}" = 'SERVER_GSSAPI_ONLY' ]]; then
+
+    DEFAULT_REALM="$(cat /etc/krb5.conf | grep default_realm | awk {'printf $3'})"
+
+    os::cmd::expect_failure_and_text 'oc login' 'No Kerberos credentials available'
+    os::cmd::expect_failure_and_text 'oc whoami' 'system:anonymous'
+
+    os::cmd::expect_failure_and_text 'oc login -u user1' "An invalid name was supplied|Can't find client principal user1@${DEFAULT_REALM} in cache collection"
```

Comment:
We want to use the same file for both Ubuntu and Fedora, and their default Kerberos configurations cause different results.